### PR TITLE
DOI 887 migrate switchboard to gh

### PR DESCRIPTION
**Pull request by royki**
- Migrate CI to GitHub Actions

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+      - name: Run Test
+        run: |
+          npm install
+          npm run test:headless
+
+      - name: Deploy on develop branch
+        if: github.ref == 'refs/heads/DOI-887_*'
+        run: |
+          rsync -e "ssh -o StrictHostKeyChecking=no" -r dist/* ${{ secrets.SB_WEB_HOST }}:${{ env.SB_APP_DIR }}/develop
```
Comment (jrhender) [inline .github/workflows/deploy.yml]: Why the checkout of the develop branch? Is it just while you were testing your CI changes?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      - name: Prepare for non-master branch
+        if: github.ref != 'refs/heads/master'
+        run: |
+          cp src/environments/environment.dev.ts src/environments/environment.ts
+          echo "Prod Env Disabled"
+          npm run build
+
+      - name: Run Test
+        run: |
+          npm install
```
Comment (jrhender) [inline .github/workflows/deploy.yml:61]: Is this `npm install` necessary if we already have the "Install npm packages" step

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
+name: Deploy
+
+on:
+  push:
+    branches: [ 'develop', 'master' ]
```
Comment (jrhender) [inline .github/workflows/deploy.yml:5]: Should run also on branches for which there is a requested PR (i.e. if merging from `branch-1` to `develop`, we should build and test `branch-1`)

**Review by jrhender — CHANGES_REQUESTED** [review]
Looks great, fantastic work 🎉 ! The use of the pre-made GitHub actions really simplifies things. A couple comments to address though and need to remove the files that aren't need:
- deploy_rsa.enc
- .travis.yml

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (jrhender) [reply .github/workflows/deploy.yml]: Already fixed 👍

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (royki) [reply .github/workflows/deploy.yml:61]: It looks like caching is not working. I tried without that. but it was failing. may be I'll check later.

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (royki) [reply .github/workflows/deploy.yml:5]: Ok.
Added, 
`pull_request: [ 'develop',  'master']`

**Review by jrhender — APPROVED** [review]

**Comment by jrhender** [comment]
Looks good @royki, thanks. Might be good to get @DawidGil-jlabs 's take as well as he is now responsible for Switchboard and has some CI experience

**Comment by jrhender** [comment]
Btw there's quite a few changes @royki 😆 . Probs good to squash this PR when merging

**Review by DawidGil-jlabs — APPROVED** [review]
Lgtm, however i do not have experience with github actions. I was configuring bitbucket pipelines.

About npm caching did you try approach from here https://betterprogramming.pub/building-angular-apps-using-github-actions-bf916b56ed0c ? This one looks similar: https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action

**Comment by royki** [comment]
> Lgtm, however i do not have experience with github actions. I was configuring bitbucket pipelines.
> 
> About npm caching did you try approach from here https://betterprogramming.pub/building-angular-apps-using-github-actions-bf916b56ed0c ? This one looks similar: https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action

Hi @DawidGil-jlabs I added the cache, please merge it and we'll see how it works.
Thanks

**Pull request merged**